### PR TITLE
Billing > Assign subaccounts to sites and/or companies

### PR DESCRIPTION
**Pull request by Ninjeneer**

### CODE REVIEW

File: src/server/rest/v1/router/api/BillingRouter.ts [hunk 1]
```diff
@@ -44,6 +44,8 @@ export default class BillingRouter {
     // ROUTES for SUB-ACCOUNTS
     // -----------------------------------
     this.buildRouteBillingCreateSubAccount();
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/router/api/BillingRouter.ts:46]: ??? I was expecting a new route for the onboarding phase

this.buildRouteBillingOnboardSubAccount();

File: src/server/rest/v1/service/BillingService.ts [hunk 2]
```diff
@@ -523,7 +523,7 @@ export default class BillingService {
     const subAccount = await BillingStorage.getSubAccountByID(tenant, filteredRequest.ID);
     UtilsService.assertObjectExists(action, subAccount, `Sub account ID '${filteredRequest.ID}' does not exist`, MODULE_NAME, 'handleActivateSubAccount', req.user);
     // Check if the sub account is already activated
-    if (!subAccount.pending) {
+    if (subAccount.status === BillingAccountStatus.ACTIVE) {
```
Comment (ClaudeROSSI) [inline src/server/rest/v1/service/BillingService.ts:526]: BillingAccountStatus.PENDING is not used anymore .

Should be used in the new endpoint for onboarding

**Review by ClaudeROSSI — CHANGES_REQUESTED** [review]
Please the endpoint **handleCreateSubAccount** to send the onboarding email and transition the sub-account status from IDLE to PENDING.

Let me know if you need more details.

File: src/server/rest/v1/router/api/BillingRouter.ts (hunk 1 above)
Comment (Ninjeneer) [reply src/server/rest/v1/router/api/BillingRouter.ts:46]: As discussed, I added the getOne and getMany endpoints to this PR. But I did not added yet the onboard endpoint as this PR is focused on company/site assignment, and I did not want to make this PR fat and hard to review.
What do you think ?

File: src/server/rest/v1/service/BillingService.ts (hunk 2 above)
Comment (ClaudeROSSI) [reply src/server/rest/v1/service/BillingService.ts:526]: > BillingAccountStatus.PENDING is not used anymore .
> 
> Should be used in the new endpoint for onboarding

ok

File: src/server/rest/v1/router/api/BillingRouter.ts (hunk 1 above)
Comment (ClaudeROSSI) [reply src/server/rest/v1/router/api/BillingRouter.ts:46]: ok ... i'll review the GET and merge that one

File: src/types/Billing.ts [hunk 3]
```diff
@@ -153,11 +153,21 @@ export interface BillingAdditionalData {
...
+export interface BillingAccount extends AuthorizationActions {
   id?: string;
   accountID: string;
   activationLink?: string;
   userID?: string;
-  pending: boolean;
+  status: BillingAccountStatus;
+}
+
+export interface BillingAccountSetting {
```
Comment (ClaudeROSSI) [inline src/types/Billing.ts:170]: I would like to rename it to

BillingAccountData

is it ok with you?

**Review by ClaudeROSSI — CHANGES_REQUESTED** [review]
I would like to rename BillingAccountSettings to BillingAccountData. Le me know ...

File: src/types/Billing.ts (hunk 3 above)
Comment (Ninjeneer) [reply src/types/Billing.ts:170]: Sure

**Pull request merged**